### PR TITLE
Feature/add flow for http

### PR DESCRIPTION
**Pull request by Iftah1**
basic structure of the flow, and changed what card is represents.
not implemented the db yet

### CODE REVIEW

File: cards_exporter/icard_exporter.py [hunk 1]
```diff
@@ -1,11 +0,0 @@
-from abc import ABC, abstractmethod
```
Comment (RYCBABAC) [inline cards_exporter/icard_exporter.py:1]: Why did you delete  this?

File: controller.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
+from typing import List, Dict
```
Comment (RYCBABAC) [inline controller.py:1]: Great class naming :)

File: controller.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+from flask import Flask, request
+from consts import *
+
+
+CARDS_SERIALIZED = Dict[str, List[Dict[str, str]]]
+
+
+class Controller:
+    def __init__(self, client_db: IClientDB):
+        self.app = Flask(__name__)
```
Comment (RYCBABAC) [inline controller.py]: Why controller properties are not private?

Comment (RYCBABAC) [inline controller.py]: Use DI, the controller shouldn't know how the app is created

Comment (RYCBABAC) [reply controller.py]: BTW what is Flask I heard about it while researching on html to pdf

File: controller.py [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        self.client_db = client_db
+        self.setup_routes()
```
Comment (RYCBABAC) [inline controller.py]: Isn't this action a bit "too much" for const'? 
Consider moving it to `__enter__`

File: controller.py [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    def setup_routes(self):
+        self.app.route("/add_card", methods=[POST])(self.add_card)
```
Comment (RYCBABAC) [inline controller.py:22]: Consider adding a route manager in `util`, so adding a route would be easier and we won't have explict routes in our controller

File: controller.py [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        self.app.route("/remove_card", methods=[POST])(self.remove_card)
+        self.app.route("/filter_cards", methods=[POST])(self.filter_cards)
+
+    def run(self):
+        self.app.run(debug=True)
+
+    def add_card(self) -> Status:
+        card = Card.generate_card(request)
```
Comment (RYCBABAC) [inline controller.py:30]: What is request? 
Why doesn't a function parameter? 
How do I validate this function is called only on http request. 
Consider renaming it to make it more explict.

Comment (RYCBABAC) [reply controller.py:30]: This comment is valid for all the http methods in this file

File: entities/card.py [hunk 7]
```diff
@@ -1,32 +1,44 @@
...
+from typing import Dict, List
+from entities.card_proprties import CardProperties
+from consts import *
 
+CARDS_SERIALIZED = Dict[str, List[Dict[str, str]]]
 
 
 @dataclass
 class Card:
-    content: str
```
Comment (RYCBABAC) [inline entities/card.py:13]: Make sure HTML files shouldn't be changed because of this removal.

File: entities/card.py [hunk 8]
```diff
@@ -1,32 +1,44 @@
...
-    type: CardType
+    card_properties: CardProperties
```
Comment (RYCBABAC) [inline entities/card.py:15]: Are you sure this is a good idea to put all the card's properties in a different class? 
I think it make the card class meaningless if it's not the one holding it's properties.

File: entities/card.py [hunk 9]
```diff
@@ -1,32 +1,44 @@
...
     design_properties: CardDesignProperties
-    date: datetime
-    version: str
-    id: UUID
 
-    def serialize_card(self) -> Dict[str, str]:
+    def card_to_dict(self) -> Dict[str, str]:
```
Comment (RYCBABAC) [inline entities/card.py]: Is this supposed to be the json serializer?
Not sure what's happening here.

File: entities/card.py [hunk 10]
```diff
@@ -1,32 +1,44 @@
...
-            "date": str(self.date),
-            "version": self.version,
-            "id": self.id
+            CONTENT: self.card_properties.content,
+            CARD_TYPE: self.card_properties.type.value,
+            DATE: str(self.card_properties.date),
+            VERSION: self.card_properties.version,
+            ID: self.card_properties.id
         }
         for key, value in serialized_card.items():
```
Comment (RYCBABAC) [inline entities/card.py:26]: Not sure this loop is nessecery it's ok to leave null on json. 
BTW after running `json.load` we get `Dict[str, Any]` so it might be relevant here.

File: entities/card.py [hunk 11]
```diff
@@ -1,32 +1,44 @@
...
+    def cards_to_dict(cards: List['Card']) -> CARDS_SERIALIZED:
+        cards_dict = {
+            CARDS: []
+        }
+        for card in cards:
+            cards_dict[CARDS].append(card.card_to_dict())
+        return cards_dict
+
+    @staticmethod
+    def generate_card(request: Request) -> 'Card':
```
Comment (RYCBABAC) [inline entities/card.py:42]: Why do this function get reqeust?
Card shouldn't know it's generated by http request, it shouldn't know how it's generated ( I think it breaks the D in SOLID but don't count on it, I just don't think it's a correct design). 
Replace it to get a json string or something of a sort.

File: entities/card_proprties.py [hunk 12]
```diff
@@ -0,0 +1,42 @@
...
+class CardProperties:
+    content: str
+    version: str
+    type: CardType
+    date: datetime
+    id: UUID
+
+    @staticmethod
+    def generate_card_properties(request: Request) -> 'CardProperties':
+        card_json = request.json
```
Comment (RYCBABAC) [inline entities/card_proprties.py:22]: Again as before, change request

File: entities/card_proprties.py [hunk 13]
```diff
@@ -0,0 +1,42 @@
...
+        return CardProperties.generate_card_from_json(card_json)
+
+    @staticmethod
+    def generate_card_from_json(card_json: Dict[str, str]) -> 'CardProperties':
```
Comment (RYCBABAC) [inline entities/card_proprties.py]: Why `'CardProperties'`? 
Python thing?

File: entities/card_proprties.py [hunk 14]
```diff
@@ -0,0 +1,42 @@
...
+        content = card_json[CONTENT]
```
Comment (RYCBABAC) [inline entities/card_proprties.py:27]: Consider "cleaning" this function a little bit... It's a bit long and defines variables that only used once

File: entities/status.py [hunk 15]
```diff
@@ -0,0 +1,6 @@
+from enum import Enum
+
+
+class Status(Enum):
```
Comment (RYCBABAC) [inline entities/status.py]: Consider changing it from verbal string status to standard http statuses (for instance 200 for success) so it'll be easy to use the API (consider the case that someone new is trying to call our API and doesn't understand why it's response status is not 200)

File: consts.py [hunk 16]
```diff
@@ -0,0 +1,11 @@
+POST = "POST"
```
Comment (RYCBABAC) [inline consts.py]: Good idea, consider moving this file to `utils` or another suitable folder.

File: consts.py [hunk 17]
```diff
@@ -0,0 +1,11 @@
...
+QUESTION = "question"
+ANSWER = "answer"
+CONTENT = "content"
+CARD_TYPE = "card_type"
+DATE = "date"
```
Comment (RYCBABAC) [inline consts.py]: General: Consider working from lower to upper layers, for instance, start with implementing the DB and the other classes used in the controller and afterwards connect them all with the controller. It'll make your work much easier and it won't leave "dead" code path (paths in code that do not work/do nothing). 
General: Try to prioritize your tasks, start with the most important tasks (for instance, inserting a card) and make sure it's finished. Only when it's done and ready, move to the next tasks with lower priority. It'll make your work more organized and again won't leave "dead" code paths.

**Review by RYCBABAC — COMMENTED** [review]
Please fix comments

File: cards_exporter/icard_exporter.py (hunk 1 above)
Comment (Iftah1) [reply cards_exporter/icard_exporter.py:1]: there was icard_exporter and icardexporter and i deleted one of them, i can change the name of the file if you want :)

File: controller.py (hunk 2 above)
Comment (Iftah1) [reply controller.py:1]: fuck off

File: controller.py (hunk 3 above)
Comment (Iftah1) [reply controller.py]: fuck forgot private and stuff
this is so irelevant (jk, will be added)

Comment (Iftah1) [reply controller.py]: yeah, will be fixed :)
flask is a simple library for http servers in python
the flask you are talking about is different i think

File: entities/card_proprties.py (hunk 13 above)
Comment (Iftah1) [reply entities/card_proprties.py]: yes, if you want to return a instance of a class in the same class, you need ' ', why? idfk

File: cards_exporter/icard_exporter.py (hunk 1 above)
Comment (RYCBABAC) [reply cards_exporter/icard_exporter.py:1]: Just validate that it still works with what I wrote and nothing breaks

File: controller.py (hunk 3 above)
Comment (RYCBABAC) [reply controller.py]: Didn't you say you wanted to use fast api?

File: entities/card_proprties.py (hunk 14 above)
Comment (Iftah1) [reply entities/card_proprties.py:27]: function a bit long, but i dont think i can clean it that much
and i like the constistency of the inputs in CardProperties

File: entities/card_proprties.py (hunk 13 above)
Comment (RYCBABAC) [reply entities/card_proprties.py]: OK python shit...

File: consts.py (hunk 16 above)
Comment (Iftah1) [reply consts.py]: fixed
